src/lib/core/trigger.h

```diff
@@ -57,6 +60,19 @@ struct trigger
 	 * or the object containing the trigger is destroyed.
 	 */
 	trigger_f0 destroy;
+	/**
+	 * Flag used if triggers runs by trigger_fiber_run function,
+	 * otherwise not used.
+	 * In trigger_fiber_run function if this flag is set,
+	 * trigger runs in fiber asynchronously.
+	 */
+	bool async;
```

Comment:
I would be glad if we didn't change struct trigger.
As I see `bool async` member is used for making lua trigger synchronous whil making C API triggers asynchronous. Note that in the last commit we make them all asynchronous so the flag will always be true.
Let's make lua triggers asynchronous at first, and the implement C API triggers. Then we won't need that flag.